plugin/src/main/java/com/denizenscript/denizen/events/world/RaidScriptEvent.java

```diff
@@ -0,0 +1,87 @@
+package com.denizenscript.denizen.events.world;
+
+import com.denizenscript.denizen.events.BukkitScriptEvent;
+import com.denizenscript.denizen.objects.EntityTag;
+import com.denizenscript.denizen.objects.LocationTag;
+import com.denizenscript.denizen.objects.PlayerTag;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import com.denizenscript.denizencore.objects.core.ListTag;
+import com.denizenscript.denizencore.objects.core.MapTag;
+import com.denizenscript.denizencore.utilities.CoreUtilities;
+import org.bukkit.Raid;
+import org.bukkit.entity.Raider;
+import org.bukkit.event.raid.RaidEvent;
+
+import java.util.UUID;
+
+public class RaidScriptEvent<T extends RaidEvent> extends BukkitScriptEvent {
+
+    // <--[language]
+    // @name Raid Event Data
+    // @group Useful Lists
+    // @description
+    // Every event related to village raids has a <context.raid> property, a MapTag wrapper around raid data (see <@link url https://hub.spigotmc.org/javadocs/spigot/org/bukkit/Raid.html>).
+    // These events are <@link event player triggers raid>, <@link event raid finishes>, <@link event raid spawns wave>, and <@link event raid stops>.
+    //
+    // The data format is as follows:
+    // location: a LocationTag of the raid's center
+    // heroes: a list of PlayerTags that have participated in the raid
+    // raiders: a list of raider EntityTags that remain in the current wave
+    // status: the current status of the raid. See <@link url https://hub.spigotmc.org/javadocs/spigot/org/bukkit/Raid.RaidStatus.html>
+    // ticks: the raid's age in ticks
```

Comment:
This should probably be turned into a duration

Reply:
Updated